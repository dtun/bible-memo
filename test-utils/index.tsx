import { render } from "@testing-library/react-native";

const AllTheProviders = ({ children }: { children: React.ReactElement }) => {
  return <>{children}</>;
};

const customRender = (ui: React.ReactElement) =>
  render(ui, { wrapper: AllTheProviders });

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { customRender as render };
