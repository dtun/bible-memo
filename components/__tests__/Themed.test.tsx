import * as React from "react";

import { render, screen } from "../../test-utils";
import { Text, View } from "../Themed";

it(`renders <Text />`, () => {
  render(<Text>Snapshot test!</Text>);

  expect(screen).toMatchSnapshot();
});

it(`renders <View />`, () => {
  render(<View />);

  expect(screen).toMatchSnapshot();
});
