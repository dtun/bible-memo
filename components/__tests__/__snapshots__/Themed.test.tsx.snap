// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Text /> 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`renders <View /> 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
/>
`;
